Library/src/main/java/com/wunderlist/slidinglayer/SlidingLayer.java

```diff
@@ -864,10 +865,14 @@ protected boolean canScroll(View v, boolean checkV, int dx, int dy, int x, int y
                 }
             }
         }
-
-        return checkV && (
+        boolean temp = checkV && (
                 (allowedDirection() == HORIZONTAL && ViewCompat.canScrollHorizontally(v, -dx) ||
-                        allowedDirection() == VERTICAL && ViewCompat.canScrollVertically(v, -dy)));
+                        allowedDirection() == VERTICAL && v.canScrollVertically( -dy)));
+        if (v instanceof RecyclerView && dy > 0) {
```

Comment:
I'd suggest we turn this into some form a flat passed in from the top of the lib, where the view is added, so that we can avoid adding a complete dependency just to do a class check. Any other ideas?

Reply:
Can you please elaborate more on this?

Reply:
I was wondering whether there was a way to obtain the same outcome without needing to let the lib know about `RecyclerView`. There are two paths that come to my mind:
1. Pass a flag into the configuration of the library (eg.: `layer.setVerticallyScrollingChildren(true)`), and use that instead of the `instanceof` check.
2. Adding the support library to dependencies and use interfaces like [`ScrollingView`](https://developer.android.com/reference/android/support/v4/view/ScrollingView) to make this work not only with `RecyclerView` but also with other implementers of the interface.

Now, mentioned all this, I'm reading at the code under `RecyclerView` and `canScrollVertically` should be sufficient to determine whether the children can scroll vertically or not. What's your specific use case, and why that does not work by itself?

Reply:
Please see this video when we scroll down recycler view, Slider view gets stuck https://www.youtube.com/watch?v=4oByYVkV-dM

And if recycler view is not scrolled down slider works with no issues: https://www.youtube.com/watch?v=cpwMz34UKwE

Reply:
Interesting. It looks like a different problem to me. These buttons don't seem to be part of the RecyclerView are they? Are these custom buttons? Regular ones should not capture de click event if you are dragging it. Or is the button header part of the recyclerView too?

Reply:
Those buttons are custom tab views.

Reply:
Ok, that's most likely the reason. What you can do to confirm this behavior is to use a built-in clickable component such as `Button` or `ImageButton`, and verify whether the sliding layer responds or not.
Once confirmed, you just need to get the delivery of touch events right inside of your components, by overriding `onInterceptTouchEvent` and only capture the event if the interaction is part of a press and not a drag initiative. You can read more about it in [here](https://developer.android.com/training/gestures/viewgroup#intercept).

Said so, if your custom component is inheriting from a built in clickable component, that should have been handled already there. What type of custom component is your tab? Happy to take a loot to the source if that could help in any way.

Reply:
can we have skype call, I can share my screen, let me know your skypeID.
Thanks

Reply:
Let's use a Hangout so that we avoid sharing private info in here. What time works best with you? I can meet all throughout the day today.

Reply:
Please let me know, I can meet today.

Reply:
Can do in the next half an hour? Otherwise, would something like 19:30 work?

Reply:
Ok Sure, My hangout ID is "chirag.agrawal@tudip.nl". 
Please ping me whenever you are free.

Reply:
I have raised another PR (https://github.com/wunderlist/android-sliding-layer-lib/pull/112).
Please have a look on it.

Reply:
Hi @JlUgia,
Please have a look at PR (https://github.com/wunderlist/android-sliding-layer-lib/pull/112). 
This will not add any extra dependency.